### ISSUE TITLE
internationalization needed on description

### ISSUE THREAD

**Issue by f3ltron** [vue-styleguidist/vue-styleguidist#553]
**The problem**

actually as the doc is never in one lang you should be able to I18n components easily

**Proposed solution**

Create subfolder for i18n. and parse description and create as many files as you have trads

**Alternative solutions**

none

**Additional context**

**Comment by elevatebart** [comment]
I see three parts in this enhancement
- [ ] allowing vue-docgen-cli to load descriptions in a different locale than the native one to replace the descriptions of the docgen object
- [ ] allowing vue-docgen-cli to generate this locale descritption file automatically
structure of the locale file
button.FR.json
```js
{
  "props": {
    "color": {
      "master": "the color for the button",
      "i18n": "la couleur du bouton"
    }
  }
}
```
- [ ] having a tool that checks updates this locale object each time the component is updated and helps documenters clearly see non-translated descriptions. If a descritpion does not correspond, the translation is removed and the description is removed.

**Comment by elevatebart** [comment]
I would host the locales files near the component files.
`src/components/button.vue` would have `src/components/button.doc.FR.json` for translations in french

**Comment by Jesus82** [comment]
I'd ask to leave the option to have all the translations in a separate file, in a "locales" folder (and not based in components). The reason is that many components may share the same translation keys.

**Comment by elevatebart** [comment]
Hello @Jesus82, 

This is an interesting suggestion.

How would you organize the file in this case?
Would you have a flat list of strings to translate?
What would be the keys to access those translations?

Keep in mind that, in order to keep context, and jsdoc response active, the main idea is to keep the English doc **in the components** themselves.

Thank you

**Comment by Jesus82** [comment]
I have it organized a little bit different, with the locales folder at the same level with the components folder, and then a file for each language. That's also how I'm planning to use my components later, if fx I want to place it in a nuxt app (with a separate file, one for each language).

**Comment by elevatebart** [comment]
This folder structure this is what I understand from your description.
It feels a little harder to navigate but easier to extract a whole folder to send for translations.

```
src
 └─components
      ├─Button
      |     ├─Button.vue
      |     └─Readme.md
      └─Input
      |     ├─Input.vue
      |     └─Readme.md
      └─Locale
           └─FR
                 ├─Button
                 |     ├─Readme.md
                 |     └─descriptions.json
                 └─Input
                       ├─Readme.md
                       └─descriptions.json

```

Did I get your scaffolding right? Did you have other suggestions?

**Comment by f3ltron** [comment]
if think it's something like this yeah but not locale, use locales. Why readme in locales ? @elevatebart

**Comment by elevatebart** [comment]
Since the readme contains the documentation that will occupy the bottom of the component sheet, it could be useful to translate it as well. It might be a challenge to know when the file is obsolete. 

My first reaction would be to use markdown comments at the end of the file. This would serve as the key for the last updated original file.

```markdown
[//]: # <automatically generated code> docgen.hash = fr5tfdc3efg765hbvt
```

When asked to check translations, we could provide the diff of the original `Readme.md` since the last change. In case of mismatch, we would ask users to confirm that translation is still correct.

**Comment by Jesus82** [comment]
I was thinking more in something like this (as it mimics how I would set the translations IRL). While in CSS it can make sense to have separate CSS's for each component, and then have some common styles, in translations I prefer to have just a file and then have subobjects for each part of the web, so I can scan all terms at the same time and see if there is something that is already translated.

It is also how nuxt structure works.
```
src
 |─components
 |   ├─Button
 |    |     ├─Button.vue
 |    |     └─Readme.md
 |   ├─Input
 |    |     ├─Input.vue
 |    |     └─Readme.md
└─Local
        └─FR.json
        └─ES.json
```

**Comment by elevatebart** [comment]
We need to translate the documentation file (Readme.md). This file is rich text using markdown syntax. It contains examples of code but mostly the explanations of what the component should be used for. This last part is very important in a design system. 

Since this markdown would have to be translated would you put it inside the Jason file? How would it look?

Bart Ledoux
Twitter: @bartledoux
Github, Gitter, Discord: @elevatebart

**Comment by Jesus82** [comment]
Yep, I would do it that way (assigning a translation key to each of the markdown texts).

**Comment by elevatebart** [comment]
@Jesus82 since JSON does not accept multi-line text how would you go about this translation?
String arrays? How would you choose the keys of each part of the markdown?

As of now it very much seems an awful lot of work just for not repeating code.

**Comment by vanpipy** [comment]
Hi, any update here? I saw the PR, and it stuck the deps?

**Comment by elevatebart** [comment]
No update for now as I have had very little time to work on open source recently.

If you feel like taking it on, I can be available for support.

If not, It might take a few more months before I get back to it.

Sorry.

**Comment by vanpipy** [comment]
> No update for now as I have had very little time to work on open source recently.
> 
> If you feel like taking it on, I can be available for support.
> 
> If not, It might take a few more months before I get back to it.
> 
> Sorry.

Never mind and thanks for your work. I want the internationalization function of the vue-styleguidist and found this PR. It is a long time and I spent little time to figure out why but lots of detail I did not know, so I ask here. Thanks for your quick reply. I wound do something I can for it :D.
